src/Native/Runtime/RuntimeInstance.cpp

```diff
@@ -949,7 +949,13 @@ EXTERN_C REDHAWK_API void* __cdecl RhAllocateThunksMapping()
         // cannot reuse it here. Now we need to create a new mapping of the thunks section in order to have 
         // more thunks
 
-        UInt8* pModuleBase = (UInt8*)PalGetModuleHandleFromPointer(pThunksTemplateAddress);
+        UInt8* pModuleBase;
+#ifdef _WASM_
```

Comment:
It may be better to change the ifdef of this whole block to `#if !defined(FEATURE_RX_THUNKS) && !defined(USE_PORTABLE_HELPERS)` instead.

Add stubs to portable.cpp if there is anything required to make things link for CppCodeGen or WASM.

Reply:
This function is already in a ```#ifndef FEATURE_RX_THUNKS``` block and the callees have handling for portable helpers  (which return null/throw appropriately), so I think this is already roughly the way you're asking for.

Reply:
I meant that the whole function should be under `#if !defined(FEATURE_RX_THUNKS) && !defined(USE_PORTABLE_HELPERS)` instead.

Reply:
(Cleaning it up later is fine if you prefer to not do it as part of this PR.)

Reply:
Now that I'm no longer touching this file in this PR, I'd rather leave it as-is. I wonder if we might even be able to make this function work with the portable runtime as long as the set of thunks doesn't need to expand.

Reply:
Portable runtime cannot contain any assembly code (static or dynamically generated), so this cannot ever work with portable runtime.

Reverse PInvokes for portable runtime have to be done using statically generated strongly typed static thunks. (This comes with limitations similar to https://developer.xamarin.com/guides/ios/advanced_topics/limitations/#Reverse_Callbacks.)